server.js

```diff
@@ -40,18 +40,16 @@ defaultClient.basePath = 'https://connect.squareupsandbox.com';
 app.post('/process-payment', async (req, res) => {
   const request_params = req.body;
 
-  // length of idempotency_key should be less than 45
-  const idempotency_key = crypto.randomBytes(22).toString('hex');
-
   // Charge the customer's card
   const payments_api = new squareConnect.PaymentsApi();
   const request_body = {
     source_id: request_params.nonce,
+    location_id: request_params.location_id,
```

Comment:
where do you add the location_id?

Reply:
that is being done on the client javascript side.  We also do that for Apple Pay examples.

Reply:
I remember we removed it for simplicity.

Reply:
It seems like there is a minimum bar for simplicity.  We should balance the desire for simplicity (underestimating the brain power of our developer customers [who are likely smarter than we are]) with the need to communicate factual or opinionated statements that make their jobs easier :-)